tests/main/apt-hooks/task.yaml

```diff
@@ -58,7 +58,18 @@ execute: |
     No apt package "aws-cli", but there is a snap with that name.
     Try "snap install aws-cli"
     
-    E: Unable to locate package aws-cli
+    ERROR="E"
+    if os.query is-ubuntu-ge 24.10; then
+        ERROR="ERROR"
+    fi
+    cat > expected << EOF
+
+    WARNING: apt does not have a stable CLI interface. Use with caution in scripts.
+    
+    No apt package "aws-cli", but there is a snap with that name.
+    Try "snap install aws-cli"
+
+    $ERROR: unable to locate package aws-cli
```

Comment:
My suggestion doesn't work literally, the previous message needs to be deleted. But I like your original solution better anyway, sorry